python/cudf/cudf/tests/test_df_protocol.py

```diff
@@ -39,15 +39,12 @@ def assert_buffer_equal(buffer_and_dtype: Tuple[_CuDFBuffer, Any], cudfcol):
         cudfcol.apply_boolean_mask(non_null_idxs),
     )
 
-    if dtype[0] != _DtypeKind.BOOL:
-        array_from_dlpack = cp.from_dlpack(buf.__dlpack__()).get()
-        col_array = cp.asarray(cudfcol.data_array_view(mode="read")).get()
-        assert_eq(
-            array_from_dlpack[non_null_idxs.to_numpy()].flatten(),
-            col_array[non_null_idxs.to_numpy()].flatten(),
-        )
-    else:
-        pytest.raises(TypeError, buf.__dlpack__)
+    array_from_dlpack = cp.from_dlpack(buf.__dlpack__()).get()
+    col_array = cp.asarray(cudfcol.data_array_view(mode="read")).get()
+    assert_eq(
+        array_from_dlpack[non_null_idxs.to_numpy()].flatten(),
+        col_array[non_null_idxs.to_numpy()].flatten(),
+    )
```

Comment:
Does this pass need to be conditioned on a cupy version?

Reply:
Sure we can. We have not historically managed supported versions for testing consistently for any dependency aside from pandas AFAIK, but cupy is generally stable enough that maybe the few places where our tests are already conditional on cupy version are sufficient to capture all the cases we've cared about so far.

Reply:
Are we able to just set a minimum cupy version in our dependencies that meets this criterion, rather than introduce conditional logic? https://github.com/rapidsai/cudf/pull/12883#issuecomment-1456397914

Reply:
Just thinking that unless (as @bdice suggests) we bump the minimum version, any environment that was previously passing will now fail these tests.